server/src/main/java/org/elasticsearch/action/admin/indices/template/put/PutIndexTemplateV2Action.java

```diff
@@ -88,6 +90,17 @@ public ActionRequestValidationException validate() {
             if (indexTemplate == null) {
                 validationException = addValidationError("an index template is required", validationException);
             }
```

Comment:
I think we should add return here, otherwise assertion below will trip or we will have NPE if assertions are disabled

Reply:
Ah you're absolutely right, great catch, my brain filled in that return statement. I think an else statement would be better as we'd likely like to fill in all the validation errors